src/Fluid/Flu_FixUp_Restrict.cpp

```diff
@@ -299,19 +303,11 @@ void Flu_FixUp_Restrict( const int FaLv, const int SonFluSg, const int FaFluSg,
 //    pressure, total energy density, and the dual-energy variable
 #     if ( MODEL == HYDRO )
 //    apply this correction only when preparing all fluid variables or magnetic field
-#     ifdef DEDT_NU
-#     ifdef MHD
-      if (  ( TVarCC & _TOTAL ) == ( _TOTAL - _DEDT_NU )  ||  ResMag  )
-#     else
-      if (  ( TVarCC & _TOTAL ) == ( _TOTAL - _DEDT_NU )  )
-#     endif
-#     else
 #     ifdef MHD
-      if (  ( TVarCC & _TOTAL ) == _TOTAL  ||  ResMag  )
+      if (  ( TVarCC & ResVar ) == ResVar  ||  ResMag  )
 #     else
-      if (  ( TVarCC & _TOTAL ) == _TOTAL  )
+      if (  ( TVarCC & ResVar ) == ResVar  )
```

Comment:
I think we can replace
```
#     ifdef MHD
      if (  ( TVarCC & ResVar ) == ResVar  ||  ResMag  )
#     else
      if (  ( TVarCC & ResVar ) == ResVar  )
#     endif
```
by
```
      if (  ( TVarCC & ResVar ) == ResVar  ||  ResMag  )
```
since `ResMag == FALSE` when disabling `MHD`.

Reply:
Updated.